### ISSUE TITLE
Leading |/& is not included in intersection/union node

### ISSUE THREAD

**Issue by bradzacher** [typescript-eslint/typescript-eslint#435]
**What code were you trying to parse?**

```ts
type Foo = (
    | A
    | B
);
```

**What did you expect to happen?**
The leading `|` character is included in the TSUnionType.

**What actually happened?**
![image](https://user-images.githubusercontent.com/7462525/56234139-7ee74980-6039-11e9-95e2-9f82fa2d9d66.png)
The moused over (underlined) TSUnionType starts after the leading `|` character.

**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `1.6.0` |
| `TypeScript`                           | `3.4.3` |

**Comment by bradzacher** [comment]
Looks like it's a bug in typescript itself.

**Comment by JamesHenry** [comment]
@bradzacher Have you reported it to them? Please could you link the issue here?

**Comment by mysticatea** [comment]
@JamesHenry https://github.com/Microsoft/TypeScript/issues/30995

**Comment by bradzacher** [comment]
https://github.com/microsoft/TypeScript/pull/31265 has been merged, so the fix should be released with the next TS version

**Comment by JamesHenry** [comment]
This fix is now available in TypeScript
